src/Symfony/Component/DependencyInjection/Definition.php

```diff
@@ -830,6 +832,56 @@ public function isAbstract()
     }
 
     /**
+     * Whether this definition is deprecated, that means it should not be called
+     * anymore.
+     *
+     * @param bool   $boolean
+     * @param string $template Template message to use if the definition is deprecated
+     *
+     * @return Definition the current instance
+     *
+     * @todo Support a suggested definition instead ?
+     *
+     * @api
+     */
+    public function setDeprecated($boolean = true, $template = 'The service %s relies on a deprecated definition. You should stop using it.')
```

Comment:
As a user, I just want to know that I shouldn't use `$container->get('some_deprecated_service')`, no?